### PR TITLE
[#349] Add all remaining tests to bazel

### PR DESCRIPTION
**Pull request by elBoberido**
## Notes for Reviewer

The PR makes the remaining tests available to bazel.

Since bazel runs the tests in parallel, some of them needed to be adjusted for isolation.

## Pre-Review Checklist for the PR Author

1. [x] Add sensible notes for the reviewer
1. [x] PR title is short, expressive and meaningful
1. [x] Relevant issues are linked in the [References](#references) section
1. [x] Every source code file has a copyright header with `SPDX-License-Identifier: Apache-2.0 OR MIT`
1. [x] Branch follows the naming format (`iox2-123-introduce-posix-ipc-example`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`[#123] Add posix ipc example`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Tests follow the [best practice for testing][testing]
1. [x] Changelog updated [in the unreleased section][changelog] including API breaking changes
1. [x] Assign PR to reviewer
1. [ ] All checks have passed (except `task-list-completed`)

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/concepts/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx2/blob/main/doc/release-notes/iceoryx2-unreleased.md

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Unit tests have been written for new behavior
- [x] Public API is documented
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

Relates to #349

### CODE REVIEW

File: iceoryx2/tests/node_tests.rs [hunk 1]
```diff
@@ -248,7 +281,7 @@ mod node {
         let number_of_creators = (SystemInfo::NumberOfCpuCores.value()).clamp(2, 1024);
         const NUMBER_OF_ITERATIONS: usize = 100;
         let barrier = Barrier::new(number_of_creators);
-        let mut config = Config::global_config().clone();
+        let mut config = generate_isolated_config();
```
Comment (elBoberido) [inline iceoryx2/tests/node_tests.rs:284]: @elfenpiff  for some reasons, the `concurrent_node_creation_and_listing_works` still fails. I tried to debug but could not find the root cause.

File: iceoryx2/tests/service_publish_subscribe_tests.rs [hunk 2]
```diff
@@ -724,6 +752,8 @@ mod service_publish_subscribe {
             .open()
             .unwrap();
 
+        // NOTE: although node_2 did specify a config with default values, since
+        // node_1 was created first, the values of that node have to be preset
```
Comment (elBoberido) [inline iceoryx2/tests/service_publish_subscribe_tests.rs:756]: @elfenpiff id I get this right?

**Comment by elBoberido** [comment]
@elfenpiff the nightly toolchain was updated and now the thread exit tests are failing on the unstable builds

File: iceoryx2-bb/elementary/tests/package_version_tests.rs [hunk 3]
```diff
@@ -15,6 +15,13 @@ use iceoryx2_bb_testing::assert_that;
 
 #[test]
 fn package_version_works() {
+    // NOTE: The test is skipped when not run with cargo but with bazel
+    //       The CI which runs with cargo ensures that the constants defined
+    //       in PackageVersion::get equal the package version.
+    if option_env!("CARGO").is_none() {
```
Comment (elfenpiff) [inline iceoryx2-bb/elementary/tests/package_version_tests.rs:21]: In `iceoryx2_bb_testing` there is a macro called `test_requires` which does this little thing for you here.

Could you please rewrite this to

```rust
fn package_version_works() {
  test_requires!(option_env!("CARGO").is_some());
  // ...
```

File: iceoryx2/tests/config_tests.rs [hunk 4]
```diff
@@ -18,6 +18,13 @@ mod node_name {
 
     #[test]
     fn config_file_settings_and_default_config_are_equal() {
+        // NOTE: The test is skipped when not run with cargo but with bazel,
+        //       since it does not work with the sandbox.
+        //       The CI which runs with cargo ensures that the settings are equal.
+        if option_env!("CARGO").is_none() {
```
Comment (elfenpiff) [inline iceoryx2/tests/config_tests.rs:24]: Could you please use here `test_requires` as well.

**Review by elfenpiff — APPROVED** [review]

**Pull request merged**